cube/src/main/java/zone/cogni/asquare/cube/sort/SkolemizedModel.java

```diff
@@ -0,0 +1,48 @@
+package zone.cogni.asquare.cube.sort;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.rdf.model.Statement;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.function.Function;
+
+public class SkolemizedModel implements Function<Model, Model> {
+
+  public SkolemizedModel(@Nonnull String skolemBaseUri) {
+    this.skolemBaseUri = skolemBaseUri;
+  }
+
+  private String skolemBaseUri;
+
+  @Override
+  public Model apply(Model model) {
+    Model result = ModelFactory.createDefaultModel();
+    List<Statement> statements = new StatementSorter().apply(model);
```

Comment:
Do the statements need to be sorted? Might give some overhead on big models

Reply:
for now: yes. it will be much more work to make it work otherwise.